### PR TITLE
Added the support to sort repos by the inactivity level

### PR DESCRIPTION
**Pull request by Ilikepizza2**
Solved issue #8 Sort repos by inactivity level.
I am attaching the screen recording below showcasing the changes.

https://user-images.githubusercontent.com/95919665/193423702-abbec0cb-8ef0-4507-a4b9-1d3c645bb25b.mp4

Thank you

### CODE REVIEW

**Comment by lyqht** [comment]
hi @Ilikepizza2, good attempt! A few feedback

1. it will be nice if the popup for this is on the right instead so that it won't block the group label column name of "last push date"

image

2. from the UI it looks like you can sort the list by ascending/ descending, but at the moment it seems to only sort it by descending? so after you click it once, subsequent clicks do not affect the order within the group

**Comment by Ilikepizza2** [comment]
Alright @lyqht thanks for your feedback!
I'll fix all the issues and get back to you :)

**Comment by Ilikepizza2** [comment]
Closes issue #8 Sort repos by inactivity level.
Hello there! I did manage to solve the issue. The screenrecording is attached:

https://user-images.githubusercontent.com/95919665/193517813-f7f6f491-ee48-497c-af9e-737b0458db6a.mp4

**Comment by lyqht** [comment]
Hello @Ilikepizza2, nice work! You almost have it down 😄 

1. **Responsive design:** i'm guessing you have been developing the app on a smaller window so you might not notice this problem but on different screens this happens: the icon doesn't stay next to the column name. 

image

image

Please help to make this style change.

2. **Strange behavior:** There's a bit of a funny behavior where the items in the group are only sorted if I move my mouse out of the component after clicking on "ascending/descending". 
  - If I click on an item after it is sorted, it does not select the correct item, and the items returns to its original order. 
  - This happens because the component didn't re-render correctly when you modify the value of `groupedOptions`.
  - Mutating the prop directly like what you are doing now is an anti-pattern and does not work. 
  - I understand it might be difficult to navigate the project, and the dropdown component looks quite complex to understand where to modify it. Hence, I will be helping you in this PR, and add a commit to address this.
  - This commit will also include the refactor work based on my review comment above which u didn't address.
3. I'm guessing you didn't run the husky hooks since quite a few files don't look formatted. I have done it in the new commit, but I would like you to add another commit to remove the unused classes such as `...`. Remember to run `pnpm format && pnpm lint` in your shell once you are done with your changes. You can refer to `package.json` to see what each of them do.

TLDR: Please fix the responsive design issue of the icon and remove unused classes. Thanks!

**Comment by Ilikepizza2** [comment]
Thanks for your detailed feedback!
I'll get to fixing it asap. By the way, what did you mean by the refactor work?

**Comment by lyqht** [comment]
Hi @Ilikepizza2, refactoring work is for combining `sortGroupAscending` and `sortGroupDescending` together as a single function, and also remove the need to check via group label, as mentioned in the review feedback I gave you on the highlighted block above.

**Comment by Ilikepizza2** [comment]
Oh... I am not able to see your review comment somehow :thinking: . Anyways, I'll fix the responsiveness and linting, etc and get back to you soon.
Thanks

**Comment by Ilikepizza2** [comment]
While making it responsive, I realized that the drop down seems a bit redundant given the svg, so I modified the svg to sort it accordingly. Screenshots attached below:
[screen-recording.webm](https://user-images.githubusercontent.com/95919665/193538835-70f8432a-f372-4d92-bb5a-f67af25c2bbe.webm)



Does it seem better now?
Screenshot_20221003_142629

**Comment by lyqht** [comment]
Oh it looks better now! If you're removing the dropdown, could you modify the top/bottom arrow to have an "active" state e.g. user has clicked on it, it should be white and the other arrow should be a dark color.

Thank you :)

**Comment by Ilikepizza2** [comment]
Alright :D

**Comment by Ilikepizza2** [comment]
Closes #18 
Finally everything done as stated: passed active states as props and implemented the requested feature. Linting and formatting done as expected. 
Hope everything is up to the mark.
Thanks for assisting me in solving this issue... I learned a lot!

**Comment by lyqht** [comment]
Screenshot before minor styling (ascending order)

image

Screenshot after minor styling (ascending order)

image

**Comment by Ilikepizza2** [comment]
Ohhh makes more sense now 😂
